mango/shap_analysis/shap_analysis.py

```diff
@@ -0,0 +1,670 @@
+import os
+from typing import Union, List
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import shap
+from sklearn.pipeline import Pipeline
+
+
+class ShapAnalyzer:
+    """
+    Class to analyze the shap values of a model.
+
+    :doc-author: baobab soluciones
+
+    Usage
+    -----
+
+    >>> from mango.shap_analysis import ShapAnalyzer
+    >>> shap_analyzer = ShapAnalyzer(
+    ...     problem_type="regression",
+    ...     model_name="model_name",
+    ...     estimator=estimator,
+    ...     data=data,
+    ...     metadata=metadata,
+    ...     shap_folder=shap_folder,
+    ... )
+    >>> shap_analyzer.summary_plot(show=True, file_path_save=file_path_save)
+    >>> shap_analyzer.bar_summary_plot(show=True, file_path_save=file_path_save)
+    >>> shap_analyzer.waterfall_plot(query="age > 30", path_save=path_save)
+    >>> shap_analyzer.partial_dependence_plot(
+    ...     feature="age",
+    ...     interaction_feature="height",
+    ...     class_name=1,
+    ...     file_path_save=file_path_save,
+    ... )
+    >>> shap_analyzer.get_sample_by_shap_value(
+    ...     shap_value=0.5,
+    ...     feature_name="age",
+    ...     class_name=1,
+    ...     operator=">=",
+    ... )
+    >>> shap_analyzer.make_shap_analysis(
+    ...     queries=["age > 30"],
+    ...     pdp_tuples=[("age", "height")],
+    ... )
+    """
+
+    def __init__(
+        self,
+        *,
+        problem_type: str,
+        model_name: str,
+        estimator: object,
+        data: Union[pd.DataFrame, np.ndarray],
+        metadata: Union[str, List[str]] = None,
+        shap_folder: str = None,
+    ):
+        # Tree Explainer models name
+        self._TREE_EXPLAINER_MODELS = [
+            "XGBRegressor",
+            "LGBMClassifier",
+            "LGBMRegressor",
+            "CatBoostClassifier",
+            "CatBoostRegressor",
+            "RandomForestClassifier",
+            "RandomForestRegressor",
+            "ExtraTreesClassifier",
+            "ExtraTreesRegressor",
+        ]
```

Comment:
I'm torn about these lists of "valid" values. I usually place them on a const.py file and import them from there. You can see an example of this in PR #99

Reply:
I have changed this